### ISSUE TITLE
PBS_Triplanar (and other) materials not casting shadows

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#576]
As reported by H3BO3, LucasRo7, and GearBell, certain objects are not casting shadows regardless of the value of ShadowCastMode.

PBSTriplanar has been identified as one such shader exhibiting this issue- on certain indoor scenes where it was used, light is passing through objects with that material applied when it wasn't previously.

It is believed that there may be other materials exhibiting this same issue as it has also been confirmed on PBS_DualSidedMetallic.

This is reportedly a fairly recent behavior as of sometime in the last few of builds.

**Comment by shiftyscales** [comment]
It has been suspected it could be related to the upgrade to Unity 2019.3.14f1 from 2019.3.13f1.

**Comment by Frooxius** [comment]
This is definitely not the Unity upgrade, that doesn't affect shaders like that. Can you get me list of the shaders that definitely do this? Is this replicable in a simple enviroment?

**Comment by shiftyscales** [comment]
To replicate- I created a box with a DevToolTip, grabbed its material with the MaterialTip, then converted the material to numerous other materials to compile the following lists:

Not working:
PBS_ColorMaskMetallic
PBS_ColorMaskSpecular* (PBS Color Mask Specular also has a strange issue where it overlays on top of inspector panels, context menu, etc.)
PBS_ColorSplatMetallic
PBS_ColorSplatSpecular
PBS_DualSidedMetallic
PBS_DualSidedSpecular
PBS_IntersectMetallic
PBS_IntersectSpecular
PBS_MultiUV_Metallic
PBS_MultiUV_Specular
PBS_RimMetallic
PBS_RimSpecular
PBS_TriplanarMetallic
PBS_TriplanarSpecular
PBS_VertexColorMetallic
PBS_VertexColorSpecular
PBS_VoronoiCrystal
PBSLerpMetallic
PBSLerpSpecular

Working:
PBS_DisplaceMetallic
PBS_DisplaceSpecular
PBS_SliceMetallic
PBS_SliceSpecular
FlatLitToon
PBS_Metallic
PBS_Specular
XiexeToonMaterial

It's possibly unrelated as it's the only material I saw that behaved this way. As noted above- PBS_ColorMaskSpecular was acting very strangely. It was rendering on top of material inspectors, context menus, etc. with a sort of banded noise.
![image](https://user-images.githubusercontent.com/54213390/82732338-7f80f380-9cc1-11ea-95fa-cdabd5c334fb.png)
